src/Factory.php

```diff
@@ -41,7 +41,7 @@ public function __construct(array $config = [], Client $client = null, array $cl
     /**
      * Returns the Orange DAM API Endpoint requested by name.
      */
-    public function getEndpoint(string $endpoint_name, mixed $args = []): Endpoint
+    private function getEndpoint(string $endpoint_name, mixed $args = []): Endpoint
```

Comment:
@apotek I would have probably taken this out and just instantiated each of the classes directly in the methods below. Is there a reason you like this approach better?

Reply:
Because I can implement the class path and error catching once.

I realize now that this is private that we don't need the error catching. I'll remove that. 

Can set the class path through a convenience method or private var. Do you want me to do that?

Reply:
Your review prompted me to radically simplify everything. There were still old assumptions and old requirements in my head. Thank you. Posting the updates now.